src/ray/object_manager/object_manager.cc

```diff
@@ -564,12 +611,20 @@ ray::Status ObjectManager::LookupRemainingWaitObjects(const UniqueID &wait_id) {
       RAY_RETURN_NOT_OK(object_directory_->LookupLocations(
           object_id,
           [this, wait_id](const ObjectID &lookup_object_id,
-                          const std::unordered_set<ClientID> &client_ids, bool created) {
+                          const std::unordered_set<ClientID> &client_ids,
+                          bool inline_object_flag,
+                          const std::vector<uint8_t> inline_object_data,
+                          const std::string &inline_object_metadata,
+                          bool created) {
             auto &wait_state = active_wait_requests_.find(wait_id)->second;
             if (!client_ids.empty()) {
               wait_state.remaining.erase(lookup_object_id);
               wait_state.found.insert(lookup_object_id);
             }
+            if (inline_object_flag) {
```

Comment:
We should always consider the object as ready if this flag is true. Similar to the callback below, we should merge this block with the above condition like this:
```
if (!client_ids.empty() || inline_object_flag) {
  ...
  if (inline_object_flag) {
    ...
  }
}
```